linker/shared/src/main/scala/org/scalajs/linker/standard/IOThrottler.scala

```diff
@@ -48,33 +47,19 @@ private[linker] final class IOThrottler(totalSlots: Int) {
   }
 
   private val onComplete: Any => Unit = { _ =>
-    slots.incrementAndGet()
+    semaphore.release()
     process()
   }
 
   private def process(): Unit = {
-    while (!queue.isEmpty() && tryGetSlot()) {
+    while (!queue.isEmpty() && semaphore.tryAcquire()) {
       val work = queue.poll()
       if (work == null) {
         // We raced. Release the slot and try again.
-        slots.incrementAndGet()
+        semaphore.release()
       } else {
         work()
       }
     }
   }
-
-  @tailrec
-  private def tryGetSlot(): Boolean = {
-    val s = slots.get()
-    if (s > 0) {
-      if (slots.compareAndSet(s, s - 1)) {
```

Comment:
I _think_ this is here because of the JVM and is the cause of the failing test; otherwise I don't quite understand how you'd get into the `else` branch?

Reply:
You get into the else branch if there is a race condition with another thread.

For example take the following execution schedule of `tryGetSlot` for `t1` and `t2`

1. `slots = 1`
2. t1: `s = slots.get() = 1`
3. t2: `s = slots.get() = 1`
4. t2: `slots.compareAndSet(1, 0) = true` (`slots = 0`)
5. t1: `slots.compareAndSet(1, 0) = false` (`slots` unchanged)

Reply:
Yes; but that makes sense only if you have more than one thread? So I guess the implementation needs to work on the JVM as well? In any case I'll add the behaviour back into the `Semaphore`.

Reply:
> Yes; but that makes sense only if you have more than one thread?

Yes. `IOThrottler` needs to work both on the JVM and on JS.

>  In any case I'll add the behaviour back into the Semaphore.

That is not necessary. The implementation of `Semaphore` is JS only.